src/components/GroupAndSearchDropdown/index.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import { Dropdown } from 'antd'
+import PropTypes from 'prop-types'
+import { Icon } from '../Icon'
+import { Text } from '../Typography'
+import { TextField } from '../Textfield/index'
+import './groupAndSearchDropdown.scss'
+
+export const GroupAndSearchDropdown = ({ options, onSearchChange, className, ...props }) => {
+  const [showDropdown, setShowDropdown] = useState(false)
+  const [value, setValue] = useState('')
+
+  const OptionsDropdown = () => {
+    return (
+      <div className="options-dropdown">
+        <div className="search-box">
+          <TextField type="search-box" placeholder="Search" onChange={onSearchChange} />
+        </div>
+        {options.map((option, index) => (
+          <>
+            <div className="group-header" key={index}>
+              <Text type="captions_bold" color="gray-2">
+                {option?.title}
+              </Text>
+            </div>
+            {option?.children?.map((child, childIndex) => (
+              <div
+                className="group-option"
+                key={childIndex}
+                onClick={() => {
+                  setValue(child)
```

Comment:
on selection, we should append the selected value to the existing value. We should not clear it.